internal/exec/engine.go

```diff
@@ -387,52 +387,36 @@ func (e *Engine) renderConfig(cfg types.Config) (types.Config, error) {
 // cfgRef.Source must not be nil
 func (e *Engine) fetchReferencedConfig(cfgRef types.Resource) (types.Config, error) {
 	// this is also already checked at validation time
-	if cfgRef.Source == nil {
+	sources := cfgRef.GetSources()
+	if len(sources) == 0 {
```

Comment:
nit: let's drop the `sources` variable assignment and in-line it in the `if` as it isn't re-used.